### PR TITLE
fix(server): correctly use starlette APIs

### PR DESCRIPTION
**Pull request by sauyon**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -119,63 +119,7 @@ skip_glob = ["typings/*", "docs/*"]
 
 [tool.pyright]
 pythonVersion = "3.10"
-include = ["bentoml", "typings"]
+include = ["bentoml"]
 exclude = ['bentoml/_version.py','bentoml/__main__.py']
-strictListInference = true
-strictDictionaryInference = true
-strictSetInference = true
```
Comment (bojiang) [inline pyproject.toml:126]: why do we remove these

File: bentoml/_internal/types.py [hunk 2]
```diff
@@ -217,7 +217,7 @@ def is_compatible_type(t1: AnyType, t2: AnyType) -> bool:
 
 @json_serializer(fields=["uri", "name"], compat=True)
 @dataclass(frozen=False)
-class FileLike(t.Generic[t.AnyStr]):
+class FileLike(t.Generic[t.AnyStr], io.IOBase, t.IO[t.AnyStr]):  # type: ignore (python IO types)
```
Comment (bojiang) [inline bentoml/_internal/types.py:220]: Is this necessary for runtime or just for typing?

Comment (sauyon) [reply bentoml/_internal/types.py:220]: Just typing. I actually realized I can use `t.cast` later so I don't think this is necessary at all anymore.

File: pyproject.toml (hunk 1 above)
Comment (sauyon) [reply pyproject.toml:126]: I wanted to go back to full strict mode and just `# type: ignore` all the unknownmembertype and missingstubs issues.

Comment (bojiang) [reply pyproject.toml:126]: Is full strict mode stricter or looser than the current?

Comment (sauyon) [reply pyproject.toml:126]: stricter

Comment (bojiang) [reply pyproject.toml:126]: I prefer not to do too much style changes till a stable code base.

**Pull request merged**